### PR TITLE
Make the manifest contain name of the coins

### PR DESCRIPTION
**Pull request by pierreneter**

### CODE REVIEW

**Comment by sindresorhus** [comment]
The plan was to just add the names manually. This way we can also include coins not listed on CMC. And we don't want the manifest to have more coins than what we support as it will be used for the website.

See https://github.com/atomiclabs/cryptocurrency-icons/issues/141

**Comment by pierreneter** [comment]
Of course, you can check the length of icons on `manifest.json` before and after my code.

**Comment by sindresorhus** [comment]
Ok. I didn't look closely. The huge diff looked like you fetched everything.

How would the workflow be to add new icons to the manifest?

**Comment by sindresorhus** [comment]
And if we are going to keep it generated, I think it would be better to use https://github.com/lukechilds/coinlist so we don't have to keep it updated on CMC API changes.

**Comment by pierreneter** [comment]
With manifest generate from CMC, if we need to add a new coin, we push to icons array, like:
```
{
  symbol: "ABC"
}
```
and run `yarn name`.
But, so, I accept with getting the name from `coinlist`, 
I think we need an id for each coin, maybe it is the current symbol?

**Comment by pierreneter** [comment]
With manifest generate from CMC, if we need to add a new coin, we push to icons array, like:
```
{
  symbol: "ABC"
}
```
and run `yarn name`.
But, so, I agree with getting the name from `coinlist`, 
I think we need an id for each coin, maybe it is the current symbol?

**Comment by pierreneter** [comment]
ok

**Comment by pierreneter** [comment]
It is done. Can you check it again?

File: scripts/manifest.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const manifest = require('../manifest.json');
+const axios = require('axios');
```
Comment (sindresorhus) [inline scripts/manifest.js]: This is unused now.

File: package.json [hunk 2]
```diff
@@ -41,9 +41,12 @@
...
 	],
 	"scripts": {
-		"name": "node scripts/icon-name.js"
+		"manifest": "node scripts/manifest.js",
+		"precommit": "npm run manifest"
 	},
 	"devDependencies": {
-		"axios": "^0.18.0"
+		"alpha-sort": "^2.0.1",
+		"axios": "^0.18.0",
```
Comment (sindresorhus) [inline package.json]: This is unused now.

File: manifest.json [hunk 3]
```diff
@@ -2,1231 +2,1231 @@
...
+            "symbol": "zil"
         },
         {
             "name": "zilla",
             "symbol": "zilla"
         },
         {
             "name": "0x",
-            "symbol": "ZRX"
+            "symbol": "zrx"
```
Comment (sindresorhus) [inline manifest.json]: Symbols should stay uppercase.

**Comment by sindresorhus** [comment]
Can you create a `MAINTAINING.md` file where you describe how to add a coin, that you first need to add the symbol to `manifest.json` and then run `npm run manifest`.

**Comment by sindresorhus** [comment]
Any reason we nest the list of currencies in an `icon` property? Why not just make it top-level?

File: package.json [hunk 4]
```diff
@@ -39,5 +39,14 @@
 		"erc20-tokens",
 		"erc20",
 		"erc721"
-	]
+	],
+	"scripts": {
+		"manifest": "node scripts/manifest.js",
+		"precommit": "npm run manifest"
```
Comment (sindresorhus) [inline package.json]: There's no built-in `precommit` hook. I think you want https://github.com/typicode/husky

Comment (pierreneter) [reply package.json]: `precommit`, i'm using the hook of git.

**Comment by pierreneter** [comment]
> Can you create a MAINTAINING.md file where you describe how to add a coin, that you first need to add the symbol to manifest.json and then run npm run manifest.

Can you write it for me? My words are not good enough.

File: package.json (hunk 4 above)
Comment (sindresorhus) [reply package.json]: There's no `precommit` hook built-in: https://docs.npmjs.com/misc/scripts

**Comment by sindresorhus** [comment]
> Can you write it for me? My words are not good enough.

Sure

**Pull request merged**